src/apps/concepts/components/ViewConceptsHeader.tsx

```diff
@@ -70,13 +70,25 @@ const ViewConceptsHeader: React.FC<Props> = ({
   gimmeAUrl,
   addConceptToDictionary,
   children,
-  sources,
+
 }) => {
   const [showSources, setShowSources] = useState(false);
-  const formatPrefferedSources = Object.entries(PREFERRED_SOURCES_VIEW_ONLY).map(([key, value]) => ({ name: key , url: value }));
+  const [preferredSources, setPreferredSources] = useState();
+useEffect(() => {
+  sources = Object.entries(PREFERRED_SOURCES_VIEW_ONLY);
+
+  if (showSources) {
+    sources.concat(sources).concat(dictionaries);
+  }
+
+  setPreferredSources(
+    sources.map(([key, value]) => ({ name: key, url: value }));
+  );
+}, [PREFERRED_SOURCES_VIEW_ONLY, showSources, sources, dictionaries])
```

Comment:
```suggestion
  useEffect(() => {
    sources = Object.entries(PREFERRED_SOURCES_VIEW_ONLY);

    if (showSources) {
      sources.concat(sources).concat(dictionaries);
    }

    setPreferredSources(
      sources.map(([key, value]) => ({ name: key, url: value }));
    );
  }, [PREFERRED_SOURCES_VIEW_ONLY, showSources, sources, dictionaries])
```

Reply:
You'll notice that in this line, we added `dictionaries` as a requirement, but there's no local value named "dictionaries". We need to add that (see next comment).